### ISSUE TITLE
Enter non-public server as anonymous

### ISSUE THREAD

**Issue by pljones** [elieserdejesus/JamTaba#894]
Hi,

The current "private" server connect requires username and password.

It uses "private" to mean "non-public" -- but there are other servers around that aren't on the public list that still need an anonymous connection.  For the "private" login, then, it would be useful to have the full options -- i.e. anonymous or username/password.  (I think it's right to keep the public servers simple.)

Thanks.

-- Peter

**Comment by elieserdejesus** [comment]
Hi @pljones ! I'm not shure if I understand correctly. 

At moment is possible connect in every server (public or non-public) using anonymous connection. 
**Are you having problems to connect in non-public servers as anonymous?**

**Comment by pljones** [comment]
The private server login:
![ninjam-privateserver](https://user-images.githubusercontent.com/1549463/30220782-ad958fd8-94b8-11e7-921c-ac48b79138bb.PNG)
This does not have the "anonymous" check box that the ReaNINJAM login allows.  That means I can't say "I'm pljones but I'm logging in anonymously to this private server".

Unless I don't understand the form.

**Comment by pljones** [comment]
OK, it's not as obvious as the ReaNINJAM UI.  You just omit the password.

**Comment by elieserdejesus** [comment]
I remember we have a discussion (me, @teamikl  and @jonjamcam, if I remember correctly) about "anon" connections. 

Our conclusions was: People playing using "anon" names is not something good, is better not stimulate this behavior. But, If somebody is really interested in the "anon" alias is always possible type "anon" in the user name field.

**Comment by pljones** [comment]
You miss the point.

There are two separate and distinct modes.

1) Connect to the server with authentication - I supply a username and the password for that username, the server checks the password and only allows access if the password is correct.  An empty password is probably not correct.

2) Connect to the server without authentication - I supply a username but I ask to connect anonymously.  The server can choose to allow or refuse this but in either case, I supply no password.

The existing UI for ReaNINJAM makes this very clear.  If you select the "Connect anonymously" checkbox, there is no password field and it requests an anonymous connection to the server.  If you deselect "Connect anonymously", a password is requested and the username/password combination is passed to the server for authentication to allow access.

The existing Jamtaba UI does not make it clear whether both modes are supported.

**Comment by elieserdejesus** [comment]
I see your point and I agree the ReaNinjam Ui is clear about the 2 mentioned connection modes. But in my opinion the problem is exactly these 2 modes: they are not clear for end users (non ninjamers), these modes are very obscure. So, **the UI is clear, but the things/concepts the UI is abstracting are not**.

I see this problem many times when trying play in ReaNinjam with non ninjamers friends. The initial connection (for my friends) was always a problem. ReaNinjam is naming the "non authenticated" concept with the term "anonymous", and I noticed this is very confusing for me (in my first ninjam experiments) and for my non ninjamer friends too. Maybe this is not clear enough for "non native english" people only, I don't know. I remember my friends complaining about the "bad and messed very first experience (the connection step)" when trying ReaNinjam first time. They just quit because the first step was not simple enough. They are non geeks, and as you know "simple" for a non geed person means "very very very simple". 

So, JamTaba is not just dropping some ReaNinjam UI feature, we are trying to abandon a "not clear enough (for new users) authentication scheme". 

If I'm not seeing the big picture here please enlighten me.

**Comment by pljones** [comment]
OK, some examples.  Could you explain how Jamtaba behaves for these:

Private Server A allows only anonymous users.
- Why am I prompted for a password?  If I supply one, should it give an error?  What is sent to the server?

Private Server B does not allow anonymous users.
- Should the UI allow an empty password?  What should be sent to the server?

Private Server C allows both anonymous users and authenticated users.
- Is this possible?  How do I let Jamtaba know whether I'm anonymous or authenticated?

**Comment by elieserdejesus** [comment]
I suppose JamTaba will show bad or incorrect behavior in your 3 questions, and the reasons are explained in my last comments. The 2 connection modes are never seriouslly considered in JamTaba. 

If you consider these auth variations are important (for the community, not for very rare and excepcional usages)  please put a checkbox in private server dialog, deactivate password field, etc.

One last thing to consider is the term "anonymous".  Maybe is a latin thing, or my bad english, but when I read something like **Why am I prompted for a password when connecting as anonymous?** my brain enter in an infinite loop 😄 .  

In my brain (maybe is my bad english) **connect as anonymous** is something like **connect without a user name**, but in ReaNinjam *anonymous* is used as **connecting without password**.  @pljones , how this sounds for you? Are you ok with these terms? The problem is just my bad english?

**Comment by pljones** [comment]
I agree, the ReaNINJAM phraseology could be better.  The logins are either "unauthenticated" (i.e. without a password, so you could be anyone at all, thus "anonymous") or "authenticated" (with password and hence an identified user).

I suppose you could have the pop-up having the password off by default with a checkbox "Server requires password" to enable it - and then send it, even if empty.

**Comment by elieserdejesus** [comment]
> I suppose you could have the pop-up having the password off by default with a checkbox "Server requires password" to enable it - and then send it, even if empty.

Sounds good to me, I will reopen the issue.
